vmm/src/lib.rs

```diff
@@ -1666,6 +1621,54 @@ impl Vmm {
     }
 }
 
+/// Configures logger using a `LoggerConfig`.
+pub fn configure_logger(logger_config: LoggerConfig, firecracker_version: &str) -> UserResult {
+    LOGGER.set_level(match logger_config.level {
+        LoggerLevel::Error => Level::Error,
+        LoggerLevel::Warning => Level::Warn,
+        LoggerLevel::Info => Level::Info,
+        LoggerLevel::Debug => Level::Debug,
+    });
+
+    LOGGER.set_include_origin(logger_config.show_log_origin, logger_config.show_log_origin);
+    LOGGER.set_include_level(logger_config.show_level);
+
+    #[cfg(target_arch = "aarch64")]
+    let options: &Vec<Value> = &vec![];
+
+    #[cfg(target_arch = "x86_64")]
+    let options = logger_config.options.as_array().unwrap();
+
+    LOGGER.set_flags(options).map_err(|e| {
+        VmmActionError::Logger(
+            ErrorKind::User,
+            LoggerConfigError::InitializationFailure(e.to_string()),
+        )
+    })?;
+    LOGGER
+        .init(
+            &AppInfo::new("Firecracker", firecracker_version),
```

Comment:
This is more of a long lasting desire, but can't we get the name of the app from cargo somehow? Similar to how we do it for the version? :)

Reply:
Tried some things, did not find a way to use a const defined in `main.rs` in another crate (in this case, `vmm`). I will think more about this.